vertexai/ImagenScreen/ImagenScreen.swift

```diff
@@ -25,29 +26,38 @@ struct ImagenScreen: View {
   var focusedField: FocusedField?
 
   var body: some View {
-    VStack {
-      TextField("Enter a prompt to generate an image", text: $viewModel.userInput)
-        .focused($focusedField, equals: .message)
-        .textFieldStyle(.roundedBorder)
-        .onSubmit {
-          onGenerateTapped()
+    ZStack {
+      VStack {
+        InputField("Enter a prompt to generate an image", text: $viewModel.userInput) {
+          Image(
+            systemName: viewModel.inProgress ? "stop.circle.fill" : "paperplane.circle.fill"
+          )
+          .font(.title)
         }
-        .padding()
+        .focused($focusedField, equals: .message)
+        .onSubmit { sendOrStop() }
 
-      Button("Generate") {
-        onGenerateTapped()
+        ScrollView {
+          let spacing: CGFloat = 10
+          LazyVGrid(columns: [
+            GridItem(.fixed(UIScreen.main.bounds.width / 2 - spacing), spacing: spacing),
+            GridItem(.fixed(UIScreen.main.bounds.width / 2 - spacing), spacing: spacing),
+          ], spacing: spacing) {
+            ForEach(viewModel.images, id: \.self) { image in
+              Image(uiImage: image)
+                .resizable()
+                .aspectRatio(contentMode: .fill)
+                .frame(width: UIScreen.main.bounds.width / 2 - spacing,
+                       height: UIScreen.main.bounds.width / 2 - spacing)
```

Comment:
nit: Should we make the aspect ratio `1:1` in `ImagenViewModel` (`generationConfig.aspectRatio = .square1x1`) to better match the thumbnails here? Alternatively we could make these `4:3` or leave it as future work to open the images full screen when tapped.

Reply:
That's a good point. I personally like the idea of "tap to view in full screen".

Reply:
(Definitely not blocking for this PR though 🚀 )